### ISSUE TITLE
First test returns error 403

### ISSUE THREAD

**Issue by Traxxel** [blizzard-net/warcraft#81]
Hi, this code (clientId and clientSecret contains my newly created Client):
```
Dim clientId As String = "..."
Dim clientSecret As String = "..."
wc = New WarcraftClient(clientId, clientSecret, ArgentPonyWarcraftClient.Region.Europe, Locale.de_DE)
Dim result As RequestResult(Of Character) = Await wc.GetCharacterAsync("Dun Morogh", "Forgi", CharacterFields.All)
```
results in 

> System.Net.Http.HttpRequestException: 'Response code 403 (Forbidden) does not indicate success. Request: https://eu.api.blizzard.com/wow/character/Dun Morogh/Forgi?&locale=de_DE&fields=achievements,appearance,feed,guild,hunter pets,items,mounts,pets,pet slots,professions,progression,pvp,quests,reputation,statistics,stats,talents,titles,audit'

What am I doing wrong?

**Comment by Traxxel** [comment]
Hm, same error when using "Character Pets Collection Summary" at https://develop.battle.net/documentation/world-of-warcraft/profile-apis

**Comment by Megadeth02** [comment]
Same thing here. Did Blizzard change something from their side?

**Comment by danjagnow** [comment]
I'm guessing that part of the problem with this particular request was that the realm "Dun Morogh" contains whitespace, and the library isn't appropriately encoding that space as "%20".  However, the larger problem is that Blizzard has made a significant change to their APIs, which is documented at [Community API Migration Status](https://develop.battle.net/documentation/world-of-warcraft/community-api-migration-status).  The old endpoints have been retired.  I'm working on adding support for all the new APIs under #70, which is targeted for the 4.0 milestone.  The work isn't hard, but there are quite a few APIs.  I'll try to pick up the pace on that work.

**Comment by Traxxel** [comment]
I am looking forward for /data/wow/pet/index which has been marked as available :)

**Comment by danjagnow** [comment]
> I am looking forward for /data/wow/pet/index which has been marked as available :)

I've got a pull request open for this now.  See #83.

**Comment by danjagnow** [comment]
You can download [4.0.0-alpha.1](https://www.nuget.org/packages/ArgentPonyWarcraftClient/4.0.0-alpha.1), which is a prerelease build of the NuGet package. After you do, change your method call as follows.  Here's the C# version I tested:

```cs
var warcraftClient = new WarcraftClient(clientId, clientSecret, Region.Europe, Locale.de_DE);
RequestResult<CharacterProfileSummary> result = await warcraftClient.GetCharacterProfileSummaryAsync("dun-morogh", "forgi", "profile-eu");
```

I think the equivalent VB.NET would be this:

```vb
wc = New WarcraftClient(clientId, clientSecret, ArgentPonyWarcraftClient.Region.Europe, Locale.de_DE)
Dim result As RequestResult(Of CharacterProfileSummary) = Await wc.GetCharacterProfileSummaryAsync("dun-morogh", "forgi", "profile-eu")
```

The pet index and individual pets can also be retrieved now:

```cs
RequestResult<PetsIndex> result = await warcraftClient.GetPetsIndexAsync("static-us");
```

```cs
RequestResult<Pet> result = await warcraftClient.GetPetAsync(39, "static-us");
```

In VB.NET, I believe it would be something like this:

```vb
Dim result As RequestResult(Of PetsIndex) = Await wc.GetPetsIndexAsync("static-eu")
```

```vb
Dim result As RequestResult(Of Pets) = Await wc.GetPetAsync(39, "static-eu")
```

There is still a lot of work to do, but it is coming along.

**Comment by Traxxel** [comment]
This works now - great!
One question: 
`The pet index and **individual pets** can also be retrieved now:`
How can I retrieve my pets? Your code above returns 1280 pets which seem to be ALL pets on that realm and not **mine** - what was my original request.
Thank you for your work!

**Comment by danjagnow** [comment]
> How can I retrieve my pets?

This is now available in the [4.0.0-alpha.2](https://www.nuget.org/packages/ArgentPonyWarcraftClient/4.0.0-alpha.2) pre-release build.  There is now a `GetCharacterPetsCollectionSummaryAsync` method.  You can retrieve the mounts collection, too.

**Comment by Traxxel** [comment]
Works, thank you!
